File: PRAC/W7/distinct_lines.pl
#!/usr/bin/perl

$n = $ARGV[0];
$read = 0;
$distinct = 0;

%seen;
while ($line = <STDIN>) {
	# transform line
	$line =~ s/ //g;
	$line =~ tr/[A-Z]/[a-z]/;

	if (exists($seen{$line})) {
		$seen{$line}++;
	} else {
		$distinct++;
		$seen{$line}++;
	}
	$read++;
	
	if ($distinct == $n) {
		print "$distinct distinct lines seen after $read lines read.\n";
		exit;
	}
}

print "End of input reached after $read lines read - $n different lines not seen.\n";

File: PRAC/W9/distinct_lines.py
#!/usr/bin/python

import sys
import re

n = sys.argv[1]
n = int(n)
read = 0
distinct = 0

seen = {}
for line in sys.stdin:
	line = re.sub(r"\s", "", line)
	line = line.lower()
	if line not in seen:
		#print("DISTINCT LINE = {}".format(line))
		seen[line] = 0
		distinct += 1
	read += 1
	if n == distinct:
		print("{} distinct lines seen after {} lines read.".format(distinct, read))
		sys.exit()

print("End of input reached after {} lines read -  {} different lines not seen.".format(read, n))



